x/evm/keeper/params.go

```diff
@@ -2,57 +2,138 @@ package keeper
 
 import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
-
 	"github.com/evmos/ethermint/x/evm/types"
 )
 
+var isTrue = []byte("0x01")
+
 // GetParams returns the total set of evm parameters.
 func (k Keeper) GetParams(ctx sdk.Context) (params types.Params) {
-	// TODO: update once https://github.com/cosmos/cosmos-sdk/pull/12615 is merged
-	// and released
-	for _, pair := range params.ParamSetPairs() {
-		k.paramSpace.GetIfExists(ctx, pair.Key, pair.Value)
+	evmDenom := k.GetEVMDenom(ctx)
+	allowUnprotectedTx := k.GetAllowUnprotectedTxs(ctx)
+	enableCreate := k.GetEnableCreate(ctx)
+	enableCall := k.GetEnableCall(ctx)
+	chainCfg := k.GetChainConfig(ctx)
+	extraEIPs := k.GetExtraEIPs(ctx)
+
+	return types.NewParams(evmDenom, allowUnprotectedTx, enableCreate, enableCall, chainCfg, extraEIPs)
+}
+
+// SetParams sets the EVM params each in their individual key for better get performance
+func (k Keeper) SetParams(ctx sdk.Context, params types.Params) error {
+	if err := params.Validate(); err != nil {
+		return err
 	}
 
-	return params
+	k.setExtraEIPs(ctx, params.ExtraEIPs)
+	k.setChainConfig(ctx, params.ChainConfig)
+	k.setEvmDenom(ctx, params.EvmDenom)
+	k.setEnableCall(ctx, params.EnableCall)
+	k.setEnableCreate(ctx, params.EnableCreate)
+	k.setAllowUnprotectedTxs(ctx, params.AllowUnprotectedTxs)
+
+	return nil
 }
 
-// SetParams sets the evm parameters to the param space.
-func (k Keeper) SetParams(ctx sdk.Context, params types.Params) {
-	k.paramSpace.SetParamSet(ctx, &params)
+// GetExtraEIPs returns the extra EIPs enabled on the chain.
+func (k Keeper) GetExtraEIPs(ctx sdk.Context) types.ExtraEIPs {
+	var extraEIPs types.ExtraEIPs
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(types.ParamStoreKeyExtraEIPs)
+	if bz == nil {
+		return extraEIPs
+	}
+	k.cdc.MustUnmarshal(bz, &extraEIPs)
+	return extraEIPs
 }
 
 // GetChainConfig returns the chain configuration parameter.
 func (k Keeper) GetChainConfig(ctx sdk.Context) types.ChainConfig {
 	var chainCfg types.ChainConfig
-	k.paramSpace.GetIfExists(ctx, types.ParamStoreKeyChainConfig, &chainCfg)
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(types.ParamStoreKeyChainConfig)
+	if bz == nil {
+		return chainCfg
+	}
+	k.cdc.MustUnmarshal(bz, &chainCfg)
 	return chainCfg
 }
 
 // GetEVMDenom returns the EVM denom.
 func (k Keeper) GetEVMDenom(ctx sdk.Context) string {
 	var evmDenom string
-	k.paramSpace.GetIfExists(ctx, types.ParamStoreKeyEVMDenom, &evmDenom)
-	return evmDenom
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(types.ParamStoreKeyEVMDenom)
+	if bz == nil {
+		return evmDenom
```

Comment:
Nit: remove unused var
```suggestion
		return ""
```